samcli/local/layers/layer_downloader.py

```diff
@@ -106,7 +107,7 @@ def download(self, layer: LayerVersion, force=False) -> LayerVersion:
             LOG.info("%s is already cached. Skipping download", layer.arn)
             return layer
 
-        layer_zip_path = layer.codeuri + ".zip"
+        layer_zip_path = f"{layer.codeuri}_{uuid.uuid4().hex}.zip"
         layer_zip_uri = self._fetch_layer_uri(layer)
         unzip_from_uri(
```

Comment:
quick question, but will we come across any issues when the Layer is unpacked into the same directory across different threads?

Reply:
Oh it should be fine. This is what we want :)